gradle/wrapper/gradle-wrapper.properties

```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip
```

Comment:
`cglib` isn't allowed to make `java.lang` deep reflection calls by default since gradle 7.5
ref: https://docs.gradle.org/7.5/userguide/upgrading_version_7.html#removes_implicit_add_opens_for_test_workers

I propose that:
- Since `cglib` with `spring-boot2` is a common use-case, add an `add-opens` option for these modules to check for regression.
- I doubt we want to use cglib going forward (especially since [java 16,17 aren't officially supported](https://github.com/cglib/cglib/issues/191) and [not maintained well anymore](https://github.com/cglib/cglib/blob/master/README.md)). Added a JRE range condition for the only cglib related unit test in the `core` module.